src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -82,32 +81,37 @@ std::optional<AnyGeometry> SpatialJoinAlgorithms::getAnyGeometry(
   AnyGeometry geometry;
   try {
     bg::read_wkt(str, geometry);
+    geometries_.push_back(std::move(geometry));
   } catch (...) {
     printWarning();
     return std::nullopt;
   }
-  return geometry;
+  return geometries_.size() - 1;  // index of the last element
 }
 
 // ____________________________________________________________________________
-double SpatialJoinAlgorithms::computeDist(const AnyGeometry& geometry1,
-                                          const AnyGeometry& geometry2) const {
-  return boost::apply_visitor(DistanceVisitor(), geometry1, geometry2) * 78.630;
+double SpatialJoinAlgorithms::computeDist(
+    const std::optional<size_t> geometryIndex1,
+    const std::optional<size_t> geometryIndex2) const {
+  return boost::apply_visitor(ClosestPointVisitor(),
+                              geometries_.at(geometryIndex1.value()),
+                              geometries_.at(geometryIndex2.value()));
```

Comment:
This function should only take two `size_t` values, the checking for `value()` belongs outside it, as there is no useful handling possible here.